redis/src/redis/endpoint.bal

```diff
@@ -0,0 +1,1478 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/java;
+
+# Represents Redis client endpoint.
+public type Client client object {
+    private ClientEndpointConfiguration clientEndpointConfig = {};
+
+    handle datasource;
+
+    # Gets called when the endpoint is being initialized during the module initialization.
+    public function __init(ClientEndpointConfiguration config) returns error? {
+        self.datasource = initClient(config);
+    }
+
+    // String operations
+    # Append a value to a key.
+    #
+    # + key - The key
+    # + value - The string value to be appended
+    # + return - Length of the string after the operation
+    public remote function append(string key, string value) returns int|error {
```

Comment:
Usually ballerina modules has their own specific set of errors. You can look for example in `ballerina-http` module.

Shall we create module-specific error(s) for this too?
For an example, there can be an error `redis:Error`, which will be returned from the redis module functions.

There can be subtypes of this error, like `ConnectionError`, `AuthenticationError`, etc.

Then we can return that module specific error here. Check in other places as well.

Reply:
Updated in a2a10ce.